### PR TITLE
Align protobuf package version and proto options with kvproto

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Align protobuf package version proto options with kvproto.

Tests:

```shell
# pwd: tipb
cd ../tidb
git diff 
diff --git a/go.mod b/go.mod
index 4473dadedd..2f61627095 100644
--- a/go.mod
+++ b/go.mod
@@ -326,4 +326,6 @@ replace (
        // Please remove these dependencies.
        sourcegraph.com/sourcegraph/appdash => github.com/sourcegraph/appdash v0.0.0-20190731080439-ebfcffb1b5c0
        sourcegraph.com/sourcegraph/appdash-data => github.com/sourcegraph/appdash-data v0.0.0-20151005221446-73f23eafcf67
+
+       github.com/pingcap/tipb => ../tipb
 )
make
CGO_ENABLED=1  GO111MODULE=on go build -tags codes  -ldflags '-X "github.com/pingcap/tidb/pkg/parser/mysql.TiDBReleaseVersion=v8.5.0-alpha-26-g6679c7449d-dirty" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBBuildTS=2024-11-01 05:41:27" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitHash=6679c7449d8edc45a635ce950f077d6ea8b58843" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitBranch=master" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBEdition=Community" ' -o bin/tidb-server ./cmd/tidb-server
Build TiDB Server successfully!
```

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

File: proto/executor.proto [hunk 1]
```diff
@@ -7,11 +7,17 @@ option java_package = "com.pingcap.tidb.tipb";
...
+
 import "gogoproto/gogo.proto";
+import "rustproto.proto";
 
 option (gogoproto.marshaler_all) = true;
 option (gogoproto.sizer_all) = true;
 option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
```
Comment (nolouch) [inline proto/executor.proto:19]: Do we really need to turn off this option? I think it's helpful for the situation in https://github.com/tikv/client-go/blob/master/internal/locate/region_request.go#L1763-L1775

**Review by nolouch — DISMISSED** [review]
rest lgtm

File: proto/executor.proto (hunk 1 above)
Comment (overvenus) [reply proto/executor.proto:19]: `goproto_sizecache_all` generates `XXX_sizecache`, but it looks like it is never used in TiDB code base.
https://github.com/search?q=org%3Apingcap%20XXX_sizecache&type=code

**Review by nolouch — APPROVED** [review]

**Pull request merged**